### PR TITLE
Better printing for classes

### LINKED ISSUE #19: Better print methods for classes
From @joshwlambert in #15:

> A print.population() method would be nice, I would make it similar to the default list printing but a bit more dense, so that all the information fits into a smaller space and would remove the attribute printing

This request is also applicable for other classes.

### PR DESCRIPTION
**Pull request by pratikunterwegs**
This PR aims to fix #19 by adding better print methods for key classes.

### CODE REVIEW

**Comment by pratikunterwegs** [comment]
Hi @joshwlambert would you like to try out the new print methods for the various classes? I'm pretty ambivalent about the overall look, and happy to implement any suggestions if you have them. I imagine this will be updated as the package develops.

**Comment by joshwlambert** [comment]
@pratikunterwegs I'm happy to review. I'll wait until the PR is live before reviewing.

**Comment by joshwlambert** [comment]
@pratikunterwegs Nice work. I didn't spot anything that requires changing. I like the look of the print methods. It is inline with what I had in mind and I think it's slightly nicer than the equivalent list printing.

I'll leave any small comments I have in the code.

File: R/intervention.R [hunk 1]
```diff
@@ -141,3 +142,62 @@ no_intervention <- function(population) {
     contact_reduction = rep(0.0, times = nrow(population$contact_matrix))
   )
 }
+
+#' Print a `intervention` object
+#'
+#' @param x A `intervention` object.
+#' @param ... Other parameters passed to [print()].
+#' @noRd
```
Comment (joshwlambert) [inline R/intervention.R:150]: Is there a reason for the `@noRd` tag here? I generally have Rd files for all exported functions.

File: R/population.R [hunk 2]
```diff
@@ -134,3 +134,105 @@ validate_population <- function(object) {
...
+print.population <- function(x, ...) {
+  format(x, ...)
+}
+
+#' Format a `population` object
+#'
+#' @param x A `population` object.
+#' @param ... Other arguments passed to [format()].
+#'
+#' @return None. Formats the `population` for printing.
```
Comment (joshwlambert) [inline R/population.R:153]: I would move the `@return` to the `print` method as this is the exported function. Also recommend changing "None..." to "Returns invisibly..." as the `format` method can assign to variables due to `invisible(x)`

**Review by joshwlambert — APPROVED** [review]

File: R/intervention.R (hunk 1 above)
Comment (pratikunterwegs) [reply R/intervention.R]: Fixed in https://github.com/epiverse-trace/epidemics/pull/34/commits/c3e0e1aba2fda472a7f1b5c1c48311da0d626f02

File: R/population.R (hunk 2 above)
Comment (pratikunterwegs) [reply R/population.R]: Fixed in https://github.com/epiverse-trace/epidemics/pull/34/commits/c3e0e1aba2fda472a7f1b5c1c48311da0d626f02

Comment (joshwlambert) [reply R/population.R]: From what I can see the `print` methods still state `@return` as "None". Possible typo in intervention.R L161.

Comment (pratikunterwegs) [reply R/population.R]: What is the correct documentation for the return type of `print()`? I have always assumed it was none - or did you mean that it should be the invisible return?

**Pull request merged**